src/System.Net.Sockets/src/System/Net/Sockets/BaseOverlappedAsyncResult.Windows.cs

```diff
@@ -95,14 +95,17 @@ private static unsafe void CompletionPortCallback(uint errorCode, uint numBytes,
                     }
                     else
                     {
+                        bool addRefedNativeOverlapped = false;
                         try
                         {
+                            asyncResult._nativeOverlapped.DangerousAddRef(ref addRefedNativeOverlapped);
```

Comment:
It protects against the SafeNativeOverlapped being disposed of while the P/Invoke is in progress; if that were to happen, the disposal would be deferred until after the call.

I've not gone through the APM stuff to analyze whether that's actually a possibility.  That could be done as part of determining if/how to rip SafeNativeOverlapped out entirely, but I don't want to do that now.  This is a measurable cost in SAEA, but there are so many other overheads in the APM implementation, it's not as important there, and I'd like to focus our efforts on the thing we can make as fast as possible.